app/presenters/publication_presenter.rb

```diff
@@ -129,9 +129,46 @@ def video_embed_url
     end
   end
 
+  def promotion_choice
+    choice = promotion_choice_details['choice']
+    if choice.empty?
+      if has_legacy_organ_promotion_details?
+        "organ_donor"
+      else
+        "none"
+      end
+    else
+      choice
+    end
+  end
+
+  def promotion_url
+    url = promotion_choice_details['url']
+    url.empty? ? legacy_organ_promotion_details['organ_donor_registration_url'] : url
+  end
```

Comment:
I can’t think of any improvements to this that don’t raise more questions. 

I was wondering whether the `frontend` app knows too much about promotions in general, and that all ‘promotion’ related data should come from the artefact (i.e. title, text, url, link_text). That would imply the promotion data would be set in `Publisher` already, but causes the issue of how to update promotions across multiple sites, once they have been set (although in theory we already have that problem as the promotion URL is set in ‘Publisher'). Also the promotion ‘data’ should probably be identical for each done page, so each promotion should come with a default URL/text.

https://github.com/alphagov/govuk_content_models/pull/374 mentions “...so that eventually we can remove the old, organ donor-specific code in favour of "any promo you want as long as it's in the allowed list of `promotion_choice`s”, so maybe it’s best to hold out for that and do a refactor then.

Would like to have another chat face-to-face on Monday, but happy to merge otherwise.